Formula/sonarqube-lts.rb

```diff
@@ -5,12 +5,13 @@ class SonarqubeLts < Formula
   sha256 "9991d4df42c10c181005df6a4aff6b342baf9be2f3ad0e83e52a502f44d2e2d8"
   license "LGPL-3.0-or-later"
 
-  # The regex below should only match the LTS release archive on the Sonarqube
-  # downloads page. This is necessary because the usual index page for releases
-  # doesn't distinguish between current and LTS releases.
+  # Upstream doesn't distinguish LTS releases in the URL or filename, so this
+  # only matches versions for the formula's current major/minor version. This
+  # won't identify a new LTS version with a different major/minor but updating
+  # the `stable` URL with the new LTS will fix the check until the next time.
   livecheck do
-    url "https://www.sonarqube.org/downloads/"
-    regex(/downloads-lts.+?href=.*?sonarqube[._-]v?(\d+(?:\.\d+)+)\.(?:zip|t)/im)
+    url "https://binaries.sonarsource.com/Distribution/sonarqube/"
+    regex(/href=.*?sonarqube[._-]v?(#{version.major_minor}(?:\.\d+)*)\.zip/i)
```

Comment:
```suggestion
    regex(/href=.*?sonarqube[._-]v?(#{Regexp.escape(version.major_minor)}(?:\.\d+)*)\.zip/i)
```

I'll defer to you whether this is unnecessary because it will work both ways (`7.9` vs `7\.9`).

Reply:
Thanks for the catch. Using `Regexp#escape` is certainly preferable when interpolated text can/will include characters that need to be escaped. We do this in strategies where we're generating a default regex and it makes sense here as well 👍